### PR TITLE
Bug-Fix: ABI inferToSlice nil testcases

### PR DESCRIPTION
**Pull request by ahangsu**
## Description

A follow-up PR from #3823, adding one moar testcase to test `inferToSlice` will error when passing a `nil` with unexpected type (i.e., type other than slice or array).

### CODE REVIEW

**Review by tzaffi — DISMISSED** [review]
LGTM

**Review by jasonpaulos — DISMISSED** [review]
Looks good

File: data/abi/abi_encode_test.go [hunk 1]
```diff
@@ -1264,4 +1265,20 @@ func TestInferToSlice(t *testing.T) {
 		require.NoError(t, err, "inferToSlice on testcase %d failed to successfully infer %v", i, test.toBeInferred)
 		require.Equal(t, test.length, len(inferredSlice), "inferToSlice on testcase %d inferred different length, expected %d", i, test.length)
 	}
+
+	// one more testcase for totally nil (with no type information) is bad, should not pass the test
+	_, err := inferToSlice(nil)
+	require.Error(t, err, "infer to Slice should not accept nil interface")
+	assert.EqualError(
```
[the remark this answers is not in the document]
Comment (michaeldiamant) [reply data/abi/abi_encode_test.go]: @ahangsu Thanks for the change in https://github.com/algorand/go-algorand/pull/3826/commits/cd7f9f7bb914ffcd629af2d20cf4939992fc8050 - it's what I had in mind when raising the thread's question.  Resolving.

**Review by michaeldiamant — APPROVED** [review]
@ahangsu Thanks for bringing in my feedback around additional test cases.  ☕

File: data/abi/abi_encode_test.go [hunk 2]
```diff
@@ -1264,4 +1264,20 @@ func TestInferToSlice(t *testing.T) {
 		require.NoError(t, err, "inferToSlice on testcase %d failed to successfully infer %v", i, test.toBeInferred)
 		require.Equal(t, test.length, len(inferredSlice), "inferToSlice on testcase %d inferred different length, expected %d", i, test.length)
 	}
+
+	// one more testcase for totally nil (with no type information) is bad, should not pass the test
+	_, err := inferToSlice(nil)
+	require.Error(t, err, "infer to Slice should not accept nil interface")
+	require.EqualError(
```
Comment (jannotti) [inline data/abi/abi_encode_test.go:1271]: We can skip the `require.Error` since it's implied by the `EqualError`, right?

Comment (ahangsu) [reply data/abi/abi_encode_test.go:1271]: yes, that is correct

File: data/abi/abi_encode_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jannotti) [reply data/abi/abi_encode_test.go]: Just a reminder, they aren't quite the same, and the naming is not very intuitive.

`assert.X` means to do the X test, like assert.Equal for example, but then _continue running the test_.  The test as a whole will be marked a failure, but the test keeps running, possibly reporting more problems.  `assert` is a good choice if you have a bunch of fairly different checks in a single Test function.

**Review by jasonpaulos — APPROVED** [review]

**Review by jannotti — APPROVED** [review]

**Review by michaeldiamant — APPROVED** [review]
@ahangsu Thanks for the effort to add test cases here.  ☕
